core/src/main/resources/org/apache/spark/ui/static/stagepage.js

```diff
@@ -243,23 +243,39 @@ function createRowMetadataForColumn(colKey, data, checkboxId) {
 }
 
 function reselectCheckboxesBasedOnTaskTableState() {
-    var allChecked = true;
-    var taskSummaryMetricsTableCurrentFilteredArray = taskSummaryMetricsTableCurrentStateArray.slice();
-    if (typeof taskTableSelector !== 'undefined' && taskSummaryMetricsTableCurrentStateArray.length > 0) {
-        for (var k = 0; k < optionalColumns.length; k++) {
-            if (taskTableSelector.column(optionalColumns[k]).visible()) {
-                $("#box-"+optionalColumns[k]).prop('checked', true);
-                taskSummaryMetricsTableCurrentStateArray.push(taskSummaryMetricsTableArray.filter(row => (row.checkboxId).toString() == optionalColumns[k])[0]);
-                taskSummaryMetricsTableCurrentFilteredArray = taskSummaryMetricsTableCurrentStateArray.slice();
-            } else {
-                allChecked = false;
-            }
-        }
-        if (allChecked) {
-            $("#box-0").prop('checked', true);
-        }
-        createDataTableForTaskSummaryMetricsTable(taskSummaryMetricsTableCurrentFilteredArray);
-    }
+  var taskSummaryHasSelected = false;
+  var executorSummaryHasSelected = false;
+  var allTaskSummaryChecked = true;
+  var allExecutorSummaryChecked = true;
+  var taskSummaryMetricsTableCurrentFilteredArray = taskSummaryMetricsTableCurrentStateArray.slice();
+  if (typeof taskTableSelector !== 'undefined' && taskSummaryMetricsTableCurrentStateArray.length > 0) {
```

Comment:
Let's try the best to keep the following code block unchanged.

Reply:
change back, some indent in this js file  is wrong， got me wrong too.